pkg/kubelet/eviction/threshold_notifier_linux.go

```diff
@@ -90,28 +90,32 @@ func NewMemCGThresholdNotifier(path, attribute, threshold, description string, h
 	}, nil
 }
 
-func getThresholdEvents(eventfd int, eventCh chan<- int) {
+func getThresholdEvents(eventfd int, eventCh chan<- struct{}, stopCh <-chan struct{}) {
 	for {
 		buf := make([]byte, 8)
 		_, err := syscall.Read(eventfd, buf)
 		if err != nil {
 			return
 		}
-		eventCh <- 0
+
+		select {
+		case eventCh <- struct{}{}:
+		case <-stopCh:
+			return
+		}
 	}
 }
 
 func (n *memcgThresholdNotifier) Start(stopCh <-chan struct{}) {
-	eventCh := make(chan int, 1)
-	go getThresholdEvents(n.eventfd, eventCh)
+	eventCh := make(chan struct{}, 1)
```

Comment:
Using an unbuffered channel can't avoid the situation too.
However, `syscall.Read(eventfd, buf)` is expected to return an error if `syscall.Close(n.eventfd)` is called.  So `getThresholdEvents` will not wait for ever if stop channel is closed.

But here the buffer for the `eventCh` is really not essential.